src/Sarif.Driver/OrderedFileSpecifier.cs

```diff
@@ -47,11 +47,6 @@ private IEnumerable<IEnumeratedArtifact> EnumeratedArtifacts()
         {
             string normalizedSpecifier = Environment.ExpandEnvironmentVariables(this.specifier);
 
-            if (Uri.TryCreate(this.specifier, UriKind.RelativeOrAbsolute, out Uri uri))
-            {
-                normalizedSpecifier = uri.GetFilePath();
-            }
-
             string filter = Path.GetFileName(normalizedSpecifier);
```

Comment:
>filter [](http://example.com/codeflow?start=19&length=6)

This code now never expands a specifier to a full path. That means that things like directory name might remain something like `..\..\`. Is this intentional? It makes me concerned that we are introducing some subtle change in behavior because we no longer fully expand.

Reply:
Not intentional.  It was puzzling me *why* we did this temporary URI parsing.  You've answered that question.  

I'm of two minds on this:
1) Restoring the absolute path normalization (w/o using Uri class) is a safeguard against introducing a regression, for the reasons you're saying.  I'm also a little worried that trying to do this could introduce a different subtle error, or partially-mask a problem we would otherwise expose by going with #2.  But on balance, I think that's the conservative option.
2) We could lean into what the rest of this change is doing.  Declare we use unmodified strings for interfacing with .NET, and Uri.GetFilePath() for logging things.  GetFilePath will clean up the relative paths for the logs, and I think we can reasonably expect .NET to not choke on them.

They're both defensible.  I'll definitely at least try #1.